src/main/java/com/cargohub/controllers/UserProfileController.java

```diff
@@ -0,0 +1,119 @@
+package com.cargohub.controllers;
+
+import com.cargohub.dto.BillingDetailsDto;
+import com.cargohub.dto.UserDto;
+import com.cargohub.models.*;
+import com.cargohub.services.UserService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@RestController
+@CrossOrigin
+@RequestMapping("/user")
+public class UserProfileController {
+
+    private UserService userService;
+    private ModelMapper modelMapper;
+
+    @Autowired
+    public UserProfileController(UserService userService, ModelMapper modelMapper) {
+        this.userService = userService;
+        this.modelMapper = modelMapper;
+
+    }
+
+    @CrossOrigin
+    @GetMapping(path = "/profile/{id}",
+            consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getUser(@PathVariable long id) {
+        try {
+            UserDto user = userService.getUserById(id);
+            if (user == null) {
+                throw new UsernameNotFoundException("User with id: " + id + " not found");
+            }
```

Comment:
please, remove  unnecessary checking for null, it already exists in usersevise...

Reply:
skipped it as I fixed it for some other methods.
This one has already been fixed.